workflow/controller/controller.go

```diff
@@ -505,19 +505,61 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 				return err
 			}
 		case labelPodCompleted:
-			_, err := pods.Patch(
+			// remove the argoproj.io/finalizer finalizer from the pod
+			p, err := wfc.getPod(namespace, podName)
```

Comment:
We don't want to do a `get pods` call here. It 2x the number of requests, and would also need a 404 check.

Let me see if i can find a way to remove a single finalizer.